core/src/por/reporter.ts

```diff
@@ -0,0 +1,118 @@
+import { Aggregator__factory } from '@bisonai/orakl-contracts'
+import { ethers } from 'ethers'
+import { Logger } from 'pino'
+import { getReporterByOracleAddress } from '../api'
+import { buildWallet, sendTransaction } from '../reporter/utils'
+import {
+  CHAIN,
+  DATA_FEED_FULFILL_GAS_MINIMUM,
+  DATA_FEED_SERVICE_NAME,
+  PROVIDER,
+  PROVIDER_URL
+} from '../settings'
+import { IAggregator, IReporterConfig } from '../types'
+import { buildTransaction } from '../worker/data-feed.utils'
+
+async function shouldReport({
+  aggregator,
+  value,
+  logger
+}: {
+  aggregator: IAggregator
+  value: bigint
+  logger: Logger
+}) {
+  const contract = new ethers.Contract(aggregator.address, Aggregator__factory.abi, PROVIDER)
+  const latestRoundData = await contract.latestRoundData()
+
+  // Check Submission Hearbeat
+  const updatedAt = Number(latestRoundData.updatedAt) * 1000 // convert to milliseconds
+  const timestamp = Date.now()
+  const heartbeat = aggregator.heartbeat
+
+  if (updatedAt + heartbeat < timestamp) {
```

Comment:
Minor comment, it would make more sense to call it `now` rather than `timestamp`.